addons/website_crm_partner_assign/controllers/main.py

```diff
@@ -332,11 +333,13 @@ def partners_detail(self, partner_id, **post):
             partner = request.env['res.partner'].sudo().browse(partner_id)
             is_website_publisher = request.env['res.users'].has_group('website.group_website_publisher')
             if partner.exists() and (partner.website_published or is_website_publisher):
+                if slug(partner) != current_slug:
+                    return request.redirect('/partners/%s' % slug(partner))
                 values = {
                     'main_object': partner,
                     'partner': partner,
                     'current_grade': current_grade,
                     'current_country': current_country
                 }
                 return request.render("website_crm_partner_assign.partner", values)
-        return self.partners(**post)
+        return request.not_found()
```

Comment:
Those should be `raise` not `return` to serve the nice 404 and not the raw 404
```suggestion
        raise request.not_found()
```
![image](https://github.com/odoo/odoo/assets/30048408/641f15d2-2fa8-461c-9c93-bf58514bbd0d)